### ISSUE TITLE
Make use of a proper and clear Licensing

### ISSUE THREAD

**Issue by Dragnucs** [quran/quran_android#434]
Hello,

Thank you for the great and kind work your are doing. Please keep up the good work.

You are currently not using a clear licensing. Even if your work open source, it is unclear what we can actually do with it. My goal is to get this beautiful application in the [F-Droid](https://f-droid.org/) FOSS only android catalog.

I suggest you first start by adding a LICENSING.md or COPYING.md page where you put your licensing or at least add a _LICENSING_ paragraph there. Just using a _Terms of Use_ paragraph in the README.me isn't good practice.

Then, the second suggestion is to separate the code base from the data. Source code should be better under A GPLv3+ or AGPLv3+. And then data can by given an appropriate Creative Commons which doesn't dishonor the Holy Quran by making benefits from it. CC-BY-ND-NC naturally applies to Quran text itself and recitations.
- Data got from [tanzil.net](http://tanzil.net/download/) are under CC-BY-ND is the only problem. But I will try to convince F-Droid team to make an exception.
- Data from [quran.com-images](https://github.com/quran/quran.com-images) are GPL which is great.
- Data from [www.everyayah.com](http://www.everyayah.com/) and thus from [versebyversequran.com](http://www.versebyversequran.com/site/licence-for-mp3s/) are CC-BY-NC because it is Quran too.

**Finally**, every thing can be GPL'd except the actual true Quran content.

Some links to help you decide choosing a license, even if I think you can't choose anything else than GPL because of the images from quran.com-images: 
- http://choosealicense.com/
- https://creativecommons.org/choose/

Thank you for your consideration.

**Comment by ahmedre** [comment]
@Dragnucs sorry for the insanely late reply - anyhow, your solution sounds reasonable to me. can you submit a patch (or at least point me to the right direction?) i know that to do one or the other, i'd just need the respective license file into the code - is there anything i need to do to have both?

**Comment by Dragnucs** [comment]
@ahmedre Sorry for my late reply too. I stopped checking github regularly.

To apply GPLv3+ License, you only need to include the COPYING file and, preferably, add the licensing notice header in each source code file.

Please see [this repo](https://github.com/baedert/corebird) which uses GPL and this [other one](https://github.com/siacs/Conversations/) which is an android application too. This last one, _conversations_, is in F-Droid, which means it is an approved free software and does what is needed to be considered so.

**Comment by ahmedre** [comment]
closing since the app is no longer open source.

**Comment by shawnl** [comment]
F-droid has a great Bible/Torah reading app, and it would be nice for it to have this great Qur'an reading app too 

https://f-droid.org/wiki/page/net.bible.android.activity

**Comment by ozbek** [comment]
I think, this should probably be addressed sooner than later. Even if not for for F-Droid, an app like this needs a clear and more apparent licensing. @ahmedre, what do you think about GPLv3+? That still would apply only for the source code. The data files are not bundled, hence, not relevant.

**Comment by ahmedre** [comment]
I am not against it - my only concerns are:
1. people not messing with the data (and it seems like that isn't a problem),
2. we are paying hosting for the servers, etc, and these licenses all allow for free commercial use - if we leave that as is, anyone can download the app, add ads all over it, and re-distribute it, costing us money (using our servers, bandwidth etc), using our name (making people think our app is spam and not use it), etc.

if you can suggest a solution for the second issue, i am good with doing this.

**Comment by ahmedre** [comment]
@Dragnucs since CC-BY-ND-NC means "you can't sell the data" (therefore you can't really sell the app by extension, right?) - should I just make Quran's license more permissive (MIT) and request that people don't use audio files from our servers?

**Comment by ozbek** [comment]
Is not MIT for source code/software and CC is for data? How/why do you want to use MIT for data?

**Comment by ahmedre** [comment]
sorry - let me clarify - the data should be CC-BY-ND-NC - completely agreed.

what i meant was, since that's the case, should I make Quran's code under MIT (with the requests that people respect the data license and don't use our servers for hosting).

**Comment by ozbek** [comment]
Ah yes, a project can have multiple licenses: code in MIT, data in CC-BY-ND-NC.

**Comment by ahmedre** [comment]
yes - so wondering if we should do MIT instead of GPLv3 for the code

**Comment by ozbek** [comment]
Yeap, let's hope @Dragnucs will be able to comment on that.

**Comment by Dragnucs** [comment]
MIT is a lax license and allows for people to fork the project and publish it under a proprietary license. They may do it just after changing the name, icon and design and get credit for the work, or they may actually improve the application but not contribute back their contributions. Since this is a great application any contribution is needed.

Publishing the application under GPLv3+ is the best way to go since it requires people to contribute back their improvements by *forcing* them to publish their work under GPL too.

Additionally, the GPL allows you to sell the application. Many application are GPL'ed and are paid-for in the play store while free/no-cost elsewhere. Or, are in two exact and same versions; a paid and a no-cost one. Allowing people to easily contribute money if they want to. A good example is the application [conversations](http://conversations.im/)

Regading the data, it is the Holy Quran, it is the work of God. It doesn't belong to any body. In a perfect world, it should be in the public domain --a free culture material. But since you want to enforce the moral obligation of not selling or altering the words of God, a BY-SA-ND is sufficient while not really a free-culture approved. BY-SA-ND-NC is more restrictive.

If I understand your concerns, you are worried about bandwidth. A simple solution is the usage of mirrors. I would be happy to host a mirror of the text and voice data. We would then dress a list of mirrors and pick one randomly/allow user choose one at the first run and use it to download. The mirrors-list can be updated from the official applications web-site in addition to the mirror itself.

In my personal opinion, the Holy words of God should be in CC0 (free domain) and everyone shall be judged for his actions by God.

To summarize, all data under CC-BY-SA (-NC/-ND) and code under GPL. Having the data downloaded **after** the application is installed would make the application free software. Even if you chose CC-BY-SA-NC, you *can* sell the applications. It doesn't restrict you, it restricts other people forking your work!

**Comment by ahmedre** [comment]
jazakAllah khairan @Dragnucs - i'll update the docs to mention that data is not ours and is under whatever license the people who own it have placed it under (usually some sort of waqf license).

**Comment by Dragnucs** [comment]
@ahmedre Thank you very much. May Allah reward you for the great work you do.

**Comment by ahmedre** [comment]
waiyyakum and jazakAllah khair for your help with this!
